resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -125,6 +125,14 @@ private[spark] class ExecutorPodsAllocator(
     newlyCreatedExecutors --= k8sKnownExecIds
     schedulerKnownNewlyCreatedExecs --= k8sKnownExecIds
 
+    // Although we are going to delete some executors due to timeout in this function,
+    // it takes undefined time before the actual deletion. Hence, we should collect all PVCs
+    // in use at the beginning. False positive is okay in this context in order to be safe.
```

Comment:
Does it mean the PVCs here include both in-use and not in-use (from removed executors)?

Reply:
Spark only gets `snapshots` here. So, the snapshot cannot be the same with K8s AS-IS status. Always, there is a delay and some deleted executors will be notified at the next snapshot by removing from the snapshot.
> both in-use and not in-use (from removed executors)?

In addition, we cannot reclaim the PVC in the `Terminating` status. It's still in the snapshot and connected to the PVCs.

Reply:
I guess this one is answered correctly.